rs/index/src/collection/collection.rs

```diff
@@ -905,6 +906,33 @@ impl<Q: Quantizer + Clone + Send + Sync + 'static> Collection<Q> {
             sequence_number,
         )
     }
+
+    pub fn auto_vacuum(&self) -> Result<()> {
+        // Loop through all the segments and filter out segments that has a high deleted ratio
+        for segment in self.all_segments.iter() {
+            let segment_name = segment.key();
+            let segment_value = segment.value();
+
+            let total_size = segment_value.size_in_bytes_immutable_segments();
+            // Continue if total size is 0
+            if total_size == 0 {
+                continue;
+            }
```

Comment:
let's have a `should_auto_vacuum` function in the segment. In the future we might change `size_in_bytes_immutable_segments` and this code will fail